### ISSUE TITLE
FAST segmentation gives error after Ants Registration of T1w to MNI

### ISSUE THREAD

**Issue by akelkar26** [ANTsX/ANTs#1215]
Hi,

I am opening this issue because I am running into a problem reproducing an ANTsRegistration call used in qsiprep via my local ANTs commands. 
I am trying to run the registration step with the parameters mentioned here:
https://github.com/PennLINC/qsiprep/blob/master/qsiprep/data/quick_syn.json

{
    "dimension": 3,
    "convergence_threshold": [ 1e-03, 1e-03, 1e-03 ],
    "convergence_window_size": [ 20, 20, 10 ],
    "metric": [ "Mattes", "Mattes", "Mattes" ],
    "metric_weight": [ 1, 1, 1 ],
    "radius_or_number_of_bins": [ 32, 32, 56 ],
    "transforms": [ "Rigid", "Affine", "SyN" ],
    "transform_parameters": [ [ 0.01 ], [ 0.08 ], [ 0.1, 3.0, 0.0 ] ],
    "number_of_iterations": [ [ 100 ], [ 50, 25, 10], [5, 2] ],
    "sampling_strategy": [ "Random", "Regular", "Regular" ],
    "sampling_percentage": [ 0.15, 0.15, 0.25 ],
    "smoothing_sigmas": [ [ 4 ], [ 4, 2, 0 ], [ 1, 0 ] ],
    "sigma_units": [ "vox", "vox", "vox" ],
    "shrink_factors": [ [ 4 ], [ 4, 2, 1 ], [ 2, 1 ] ],
    "use_estimate_learning_rate_once": [ true, true, true ],
    "use_histogram_matching": [ true, true, true ],
    "collapse_output_transforms": true,
    "write_composite_transform": false,
    "output_transform_prefix": "ants_t1_to_mni",
    "output_warped_image": true,
    "interpolation": "LanczosWindowedSinc"
}

My command is below.
It gives me the outputs with no errors (see attached text file) but it runs super fast (likely due to the number of iterations).
However, when I use the registered output to do N4 bias field correction and compute the tissue types using fast (for the T1w in the MNI space) it gives me an error at the fast step: Exception: Not enough classes detected to init KMeans

It made me think the registration is wrong despite the output images looking fine I might be giving it the wrong parameters as I try to match it with the qsiprep call. I am not able to figure that out. Could you please help?  

Thank you in advance for all the help. 

Best,
Apoorva

Command:

#T1w to MNI space Registration                                                                                                                                                                                                                  t1brain=/Users/neurouser1/Documents/TMS_Aphasia/DWIPreprocPipelineTest2/sub-${sub}/dwi/sub-${sub}_desc-preproc_T1w.nii.gz

template=/Users/neurouser1/Documents/TMS_Aphasia/ANTsAnatPipelineTest2/sub-${sub}/anat/sub-${sub}_T1w_MNIbrainBrainExtractionBrain.nii.gz                                                                                                                                                                                                                               thisfolder=/Users/neurouser1/Documents/TMS_Aphasia/ANTsAnatPipelineTest2/sub-${sub}/anat

antsRegistration --dimensionality 3 --output [${thisfolder}/${sub}_ants_t1_to_mni,${thisfolder}/${sub}_ants_t1_to_mni_Warped.nii.gz] --write-composite-transform 0 --collapse-output-transforms 1 --interpolation LanczosWindowedSinc --winsorize-image-intensities [0.005,0.995] --initial-moving-transform [${template},${t1brain},1] --transform Rigid[0.01] --use-histogram-matching 1 --metric Mattes[${template},${t1brain},1,32,Random,0.15] --convergence [100,1e-03,20] --shrink-factors 4 --smoothing-sigmas 4vox --use-estimate-learning-rate-once 1 --transform Affine[0.08] --use-histogram-matching 1 --metric Mattes[${template},${t1brain},1,32,Regular,0.15] --convergence [50x25x10,1e-03,20] --shrink-factors 4x2x1 --smoothing-sigmas 4x2x0vox --use-estimate-learning-rate-once 1 --transform SyN[0.1,3,0] --use-histogram-matching 1 --metric Mattes[${template},${t1brain},1,56,Regular,0.25] --convergence [5x2,1e-3,10] --shrink-factors 2x1 --smoothing-sigmas 1x0vox  --verbose 1  

 * The full verbose output printed to the terminal when you run the command. If
   this is very long, please paste into a text file and upload as an attachment.
[ANTsRegistration_output.docx](https://github.com/ANTsX/ANTs/files/6894599/ANTsRegistration_output.docx)
 
 * Example input data that goes with the command line you provide.

t1brain:
[sub-TMSA02_desc-preproc_T1w.nii.gz](https://github.com/ANTsX/ANTs/files/6894630/sub-TMSA02_desc-preproc_T1w.nii.gz)

template:
[sub-TMSA02_T1w_MNIbrainBrainExtractionBrain.nii.gz](https://github.com/ANTsX/ANTs/files/6894634/sub-TMSA02_T1w_MNIbrainBrainExtractionBrain.nii.gz)


**System information (please complete the following information)**
Many issues are specific to a particular system. Please include all information
about your computing environment.

 - OS: Mac OS
 - Type of system: Desktop (using Linux via Terminal for bash scripting).
 
**ANTs version information**
 - ANTs code version: ANTs Version: 2.3.5.dev224-g5f1fa
Compiled: Mar  5 2021 23:24:29
 - ANTs installation type: [Compiled from source]

**Comment by ntustison** [comment]
This appears to be a question for the FSL (or QSIprep) folk as the error isn't thrown during the registration call and "the output images looking fine".  Please reopen the issue if you can reproduce the error using only ANTs calls.
